### PR TITLE
Update husky to v5, add lint-staged

### PR DESCRIPTION
**Pull request by ggwadera**
- Updated [`husky`](https://github.com/typicode/husky) to version 5.
  - [Summary of changes](https://blog.typicode.com/husky-5/)
    - Git hooks are much faster.
    - Optional install through the `postinstall` or `prepare` script in `package.json`.
    - Hooks are now configured in their corresponding files (like `.husky/pre-commit`) instead of in `package.json`
  - Changed the hooks scripts to lint/format using `lint-staged`.

- Installed [`lint-staged`](https://github.com/okonet/lint-staged)
  - `lint-staged` run linters against staged git files and helps keep the code formatted correctly when commiting.
  - Instead of running the linter against the whole project to check formatting like we used to, `lint-staged` makes it run only on the staged files making it faster.
  - Added automatic lint/formatting to `*.{js,jsx,ts,tsx}`, `*.{json,md,scss}` and `*.svg` for using svg optimizer.
  - Added some more overrides on `.prettierrc.js` to make prettier work with markdown and json.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -122,20 +124,17 @@
...
     ]
   },
-  "husky": {
-    "hooks": {
-      "pre-commit": "npm run lint && npm run test && npm run type-check",
-      "pre-push": "npm run lint && npm run test && npm run type-check"
-    }
+  "lint-staged": {
+    "*.{js,jsx,ts,tsx}": "eslint --fix",
+    "*.{json,md,scss}": "prettier --write",
```
Comment (Ulisseus) [inline package.json:137]: Does it support `mdx` files formatting?

File: .husky/pre-push [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+. "$(dirname "$0")/_/husky.sh"
+
+npm run type-check && npm test
```
Comment (Ulisseus) [inline .husky/pre-push:4]: Currently there is lint check in pre push. Can dropping it cause any problems?

File: package.json (hunk 1 above)
Comment (ggwadera) [reply package.json:137]: I'm not sure. I'll look into it.

File: .husky/pre-push (hunk 2 above)
Comment (ggwadera) [reply .husky/pre-push:4]: The linter will run on the files you have staged when you commit, so I don't see the need to run before pushing, only if you skip the hook when commiting.

File: package.json (hunk 1 above)
Comment (ggwadera) [reply package.json:137]: There's a mdx plugin for eslint that we can use

https://github.com/mdx-js/eslint-mdx

Maybe you could include in your mdx PR.

**Review by Ulisseus — APPROVED** [review]

**Review by songz — APPROVED** [review]

**Review by kondanna — APPROVED** [review]

**Pull request merged**